internal/pkg/service/stream/keboolasink/bridge/table.go

```diff
@@ -0,0 +1,104 @@
+package bridge
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/encoding/json"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/ctxattr"
+	serviceError "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/rollback"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+func (b *Bridge) ensureTableExists(ctx context.Context, tableKey keboola.TableKey, mapping table.Mapping) error {
+	// Create bucket if not exists
+	if err := b.ensureBucketExists(ctx, tableKey.BucketKey()); err != nil {
+		return err
+	}
+
+	// Get table
+	tab, err := b.getTable(ctx, tableKey)
+
+	// Create table
+	columnsNames := mapping.Columns.Names()
+	primaryKey := mapping.Columns.PrimaryKey()
+	var apiErr *keboola.StorageError
+	if errors.As(err, &apiErr) && apiErr.ErrCode == "storage.tables.notFound" {
```

Comment:
Wouldn't it be better to have error messages from `storage` API somewhere else as constanst/vars ?

Reply:
If we had a list of all the errors that it can return, I would put it as constants to `go-client`, but I don't think there is anything like that list.

So you have to try it and note what error it returns.

Reply:
@Matovidlo -> ask in `#team-connection` for list of all error codes